### DISCUSSION TITLE
How to pass the base url and authentication details to the connector in non-Laravel apps?

### DISCUSSION THREAD

**Discussion by iamniels** [saloonphp/saloon#57] [Q&A] [answered]
I'm planning on using Saloon to create a lib to interact with my own app. The lib can be used by other PHP (not Laravel) apps (consumer of the lib). How do I pass the base url and authentication details to the connector in the consumer app?

Extending a base connector and setting the name of that connector in each request in the consumer app is not efficient.

Great work BTW!

**Comment by Sammyjo20** [comment, marked as answer]
Thank you for the feedback!

In my libraries/SDKs that use Saloon, I tend to define the base URL / authentication into the constructor of your SDK/Library class. Then you can create a new instance of a connector and pass it into the connector via a constructor. 

For authentication, you could accept an instance of `AuthenticatorInterface` or create one yourself and then authenticate the connector with `$connector->withAuth()`.

Then you can re-use the same connector multiple times for the requests.

**Reply by iamniels** [reply]
Thanks!

Probably good to mention that I missed this topic in the docs: https://docs.saloon.dev/the-basics/sending-requests/sdk-style-connectors

**Comment by f4llenz** [comment]
This is what I do in my Connectors:

```
public function __construct(
    string $baseUrl;
)
{}

public function defineBaseUrl(): string
{
    return $this->baseUrl;
}
```
